components/identity-mgt-new/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/dao/impl/ConfigurationDAOImpl.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.identity.mgt.dao.impl;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.base.IdentityException;
+import org.wso2.carbon.identity.core.persistence.JDBCPersistenceManager;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+import org.wso2.carbon.identity.mgt.beans.TenantConfigBean;
+import org.wso2.carbon.identity.mgt.dao.ConfigurationDAO;
+import org.wso2.carbon.identity.mgt.dao.IdentityMgtDBQueries;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class ConfigurationDAOImpl implements ConfigurationDAO {
+
+    Log log = LogFactory.getLog(ConfigurationDAOImpl.class);
+
+    @Override
+    public TenantConfigBean getConfigurations(int tenantId) {
+        Connection connection = null;
+        PreparedStatement prepStmt = null;
+        ResultSet results = null;
+        try {
+            connection = JDBCPersistenceManager.getInstance().getDBConnection();
+            prepStmt = connection.prepareStatement(IdentityMgtDBQueries.LOAD_CONFIG_DATA);
+            prepStmt.setInt(1, tenantId);
+            results = prepStmt.executeQuery();
+            results.next();
```

Comment:
Check some of other components how JDBC access is typically done. next() is called within an if condition or while condition. Only if it evaluates to true we do the rest.